src/EditorFeatures/CSharpTest/Completion/CompletionProviders/OperatorIndexerCompletionProviderTests.cs

```diff
@@ -0,0 +1,960 @@
+                CompletionItem.Create("SomeText0"),
+                castCompletionItem,
+                CompletionItem.Create("SomeText1"),
+                CompletionItem.Create("\uffdcStartWith_FFDC_Identifier"), // http://www.fileformat.info/info/unicode/char/ffdc/index.htm
+                CompletionItem.Create("SomeText2"),
+                CompletionItem.Create("\uD884\uDF4AStartWith_3134A_Identifier"), // http://www.fileformat.info/info/unicode/char/3134a/index.htm
+                CompletionItem.Create("SomeText3"),
+            };
+            Array.Sort(completionList);
+            Assert.Collection(completionList,
+                c => Assert.Equal("SomeText0", c.DisplayText),
+                c => Assert.Equal("SomeText1", c.DisplayText),
+                c => Assert.Equal("SomeText2", c.DisplayText),
+                c => Assert.Equal("SomeText3", c.DisplayText),
+                c => Assert.Equal("\uD884\uDF4AStartWith_3134A_Identifier", c.DisplayText),
+                c => Assert.Equal("\uffdcStartWith_FFDC_Identifier", c.DisplayText),
+                c =>
+                {
+                    Assert.Same(c, castCompletionItem);
+                    Assert.Equal("(float)", c.DisplayText);
+                    Assert.Equal("\uFFFDfloat", c.SortText);
+                    Assert.Equal("float", c.FilterText);
+                });
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.Completion)]
+        [WorkItem(47511, "https://github.com/dotnet/roslyn/issues/47511")]
+        public async Task ExplicitUserDefinedConversionIsSuggestedAfterDot()
+        {
+            await VerifyItemExistsAsync(@"
+public class C
+{
+    public static explicit operator float(C c) => 0;
+}
+
+public class Program
+{
+    public void Main()
+    {
+        var c = new C();
+        c.$$
+    }
+}
+", "(float)");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.Completion)]
+        [WorkItem(47511, "https://github.com/dotnet/roslyn/issues/47511")]
+        public async Task ExplicitUserDefinedConversionIsSuggestedIfMemberNameIsPartiallyWritten()
+        {
+            await VerifyItemExistsAsync(@"
+public class C
+{
+    public void fly() { }
+    public static explicit operator float(C c) => 0;
+}
+
+public class Program
+{
+    public void Main()
+    {
+        var c = new C();
+        c.fl$$
+    }
+}
+", "(float)");
+        }
+
+        [Theory, Trait(Traits.Feature, Traits.Features.Completion)]
+        [WorkItem(47511, "https://github.com/dotnet/roslyn/issues/47511")]
+        [InlineData("c.$$", true)]
+        [InlineData("c.fl$$", true)]
+        [InlineData("c.  $$", true)]
+        [InlineData("c.fl  $$", false)]
+        [InlineData("c.($$", false)]
+        [InlineData("c$$", false)]
+        [InlineData(@"""c.$$", false)]
+        [InlineData("c?.$$", true)]
+        [InlineData("((C)c).$$", true)]
+        [InlineData("(true ? c : c).$$", true)]
+        [InlineData("c.$$ var x=0;", false)]
```

Comment:
This is bad. The SymbolCompletionProvider can make suggestions here. I think the IRecommendationService does a good job to figure out such constructs. (In this case, the syntax is parsed as `QualifiedNameSyntax` and `c` is treated as a type instead of a variable.